subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/StronglyTypedConfigurationAttributesResolveIntegrationTest.groovy

```diff
@@ -363,4 +369,153 @@ class StronglyTypedConfigurationAttributesResolveIntegrationTest extends Abstrac
         executedAndNotSkipped ':b:foo2Jar'
         notExecuted ':b:fooJar', ':b:barJar', ':b:bar2Jar'
     }
+
+    def "can select configuration thanks to producer schema disambiguation"() {
+        given:
+        file('settings.gradle') << "include 'a', 'b'"
+        buildFile << """
+            $typeDefs
+
+            project(':b') {
+               configurationAttributesSchema {
+                  matchStrictly(buildType)
+                  setMatchingStrategy(flavor, new AttributeMatchingStrategy<String>() {
+                       boolean isCompatible(String req, String can) { req == can }
+                       public <K> List<K> selectClosestMatch(AttributeValue<String> value, Map<K, String> compatibleValues) {
+                            value.whenPresent {
+                                compatibleValues.keySet() as List
+                            } getOrElse {
+                                [compatibleValues.entrySet().sort { it.value }.first().key ]
+                            }
+                       }
+                  })
+               }
+            }
+
+            project(':a') {
+                configurations {
+                    compile.attributes($debug)
+                }
+                dependencies {
+                    compile project(':b')
+                }
+                task check(dependsOn: configurations.compile) {
+                    doLast {
+                       assert configurations.compile.collect { it.name } == ['b-foo.jar']
+                    }
+                }
+            }
+            project(':b') {
+                configurations {
+                    foo.attributes($free, $debug)
+                    bar.attributes($paid, $debug)
+                }
+                task fooJar(type: Jar) {
+                   baseName = 'b-foo'
+                }
+                task barJar(type: Jar) {
+                   baseName = 'b-bar'
+                }
+                artifacts {
+                    foo fooJar
+                    bar barJar
+                }
+            }
+
+        """
+
+        when:
+        run ':a:check'
+
+        then:
+        executedAndNotSkipped ':b:fooJar'
+        notExecuted ':b:barJar'
+    }
+
+    def "both dependencies will choose the same default value"() {
```

Comment:
It will not match, because the identity of an attribute is the `(name, Type)` pair.